flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/DeclarativeScheduler.java

```diff
@@ -225,6 +227,39 @@ public DeclarativeScheduler(
         this.scaleUpController = new ReactiveScaleUpController(configuration);
     }
 
+    /**
+     * Check that the passed job graph is a streaming job.
+     *
+     * @param jobGraph
+     */
+    private void ensureStreamingJobGraph(JobGraph jobGraph) throws RuntimeException {
+        if (jobGraph.getJobType() == JobType.BATCH) {
+            throw new RuntimeException(
+                    "Jobs with JobType.BATCH are not supported by the declarative scheduler");
+        }
+        if (jobGraph.getScheduleMode() == ScheduleMode.LAZY_FROM_SOURCES_WITH_BATCH_SLOT_REQUEST) {
+            throw new RuntimeException(
+                    "Jobs with ScheduleMode.LAZY_FROM_SOURCES_WITH_BATCH_SLOT_REQUEST are not supported by the declarative scheduler");
+        }
+        jobGraph.getVertices()
```

Comment:
you could maybe reduce the indentation insanity here by doing something like

```
.map(vertex::getInputs)
.flatMap(Collection::stream)
.filter(!jobEdge.getSource().getResultType().isPipelined())
.findAny()
.ifPresent(ignored -> throw ...)
```
or, doing 2 for-loops.

Reply:
ok, I'll revisit

Reply:
I tried following your suggestion, but I didn't manage to substantially reduce the indentation insanity. I timeboxed my efforts. Let's maybe do this nicer when we contribute it to Flink.